internal/host/validator.go

```diff
@@ -1432,3 +1432,40 @@ func (v *validator) printNonOverlappingSubnets(c *validationContext, status Vali
 	}
 	return fmt.Sprintf("Unexpected status %s", status)
 }
+
+func (v *validator) isVSphereDiskUIDEnabled(c *validationContext) ValidationStatus {
+	if c.inventory == nil {
+		return ValidationPending
+	}
+
+	if c.cluster == nil {
+		return ValidationPending
+	}
+
+	if c.cluster.Platform == nil || c.cluster.Platform.Type == nil || *c.cluster.Platform.Type != models.PlatformTypeVsphere {
+		return ValidationSuccess
+	}
+
+	if c.inventory.Disks == nil {
+		return ValidationPending
+	}
+
+	for _, disk := range c.inventory.Disks {
+		if v.hwValidator.IsHDDevice(disk) && !disk.HasUUID {
```

Comment:
Why do we only look at disks that are "HD" (whatever that means, I'm not sure yet)?

Also you should probably ignore all non-eligible disks in this loop, and you'll get this "HD" check for free

Reply:
eligibility has more criterias because it used for installation but here i check not only installation device but storage devices.

Reply:
Yeah I could guess, but please explain this in the commit message / code / documentation - it's not obvious